src/app/SharpRaven/Data/SentryRequestFactory.cs

```diff
@@ -1,271 +1,288 @@
+                }
+                else
+                {
+                    string body = String.Empty;
+
+                    using (MemoryStream stream = new MemoryStream())
+                    {
+                        HttpContext.Request.InputStream.Seek(0, SeekOrigin.Begin);
+                        HttpContext.Request.InputStream.CopyTo(stream);
+                        body = Encoding.UTF8.GetString(stream.ToArray());
+                    }
+
+                    return body;
+                }
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception);
+            }
+
+            return null;
+        }
+
+
+        private static IDictionary<string, string> Convert(Func<dynamic, NameObjectCollectionBase> collectionGetter)
+        {
+            if (!HasHttpContext)
+                return null;
+
+            IDictionary<string, string> dictionary = new Dictionary<string, string>();
+
+            try
+            {
+                var collection = collectionGetter.Invoke(HttpContext);
+                var keys = Enumerable.ToArray(collection.AllKeys);
+
+                foreach (object key in keys)
+                {
+                    if (key == null)
+                        continue;
+
+                    string stringKey = key as string ?? key.ToString();
+
+                    // NOTE: Ignore these keys as they just add duplicate information. [asbjornu]
+                    if (stringKey.StartsWith("ALL_") || stringKey.StartsWith("HTTP_"))
+                        continue;
+
+                    var value = collection[stringKey];
+                    string stringValue = value as string;
+
+                    if (stringValue != null)
+                    {
+                        // Most dictionary values will be strings and go through this path.
+                        dictionary.Add(stringKey, stringValue);
+                    }
+                    else
+                    {
+                        // HttpCookieCollection is an ugly, evil beast that needs to be treated with a sledgehammer.
+
+                        try
+                        {
+                            // For whatever stupid reason, HttpCookie.ToString() doesn't return its Value, so we need to dive into the .Value property like this.
+                            dictionary.Add(stringKey, value.Value);
+                        }
+                        catch (Exception exception)
+                        {
+                            dictionary.Add(stringKey, exception.ToString());
+                        }
+                    }
+                }
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception);
+            }
+
+            return dictionary;
+        }
+
+
+        private static void TryGetHttpContextPropertyFromAppDomain()
+        {
+			if (HttpContext != null)

```

Comment:
changed this to not use the property as this would cause an infinite loop/out of memory exception is doing so.